src/Entities/BaseEntity.php

```diff
@@ -33,6 +33,13 @@ public function index(): Collection
         return $this->map($rawData);
     }
 
+    public function filteredIndex(string $parameters): Collection
```

Comment:
I'd probably just merge this into the `index` method and making the parameters optional.